src/foam/lib/json/Outputter.java

```diff
@@ -111,15 +111,53 @@ public void outputString(String s) {
   }
 
   public String escape(String s) {
-    return s.replace("\\", "\\\\")
+    s = removeColour(s);
+    s = s.replace("\\", "\\\\")
             .replace("\"", "\\\"")
             .replace("\t", "\\t")
             .replace("\r","\\r")
             .replace("\n","\\n");
+    s = removeControlCharacters(s);
+    return s;
   }
 
   public String escapeMultiline(String s) {
-    return s.replace("\\", "\\\\");
+    s = removeColour(s);
+    s = s.replace("\\", "\\\\");
+    s = removeControlCharacters(s);
+    return s;
+  }
+
+  // TODO: This does not cover all possible ANSI escape codes yet
+  public String removeColour(String s) {
```

Comment:
I don't think we should remove colours. I should be able to output any valid string to JSON. We should escape codes like \u2605 instead of just stripping.
See: https://www.crockford.com/mckeeman.html

Reply:
That makes sense, I can see how ANSI escapes shouldn't get special treatment here. That brings two questions to mind:

- Should logDAO allow ANSI escapes? (I vote yes because that opens cool possibilities)
- Should foam.core.String have meta properties to advise users (i.e. views) how to render and process them? I'm thinking like `{ class: 'String', name: 'message', meta: { ansicolors: true }`
  - Alternatively, maybe 'StringANSI' is an alias of 'String' which views can treat differently